gwt/src/main/java/org/pentaho/ui/database/event/DataHandler.java

```diff
@@ -317,10 +317,12 @@ public void loadConnectionData() {
 
   @Bindable
   public void setAzureSqlDBAuthRelatedFieldsVisible() {
-    passwordBox.setDisabled( azureSqlDBJdbcAuthMethod != null
-        && ( "Azure Active Directory - Universal With MFA".equals( azureSqlDBJdbcAuthMethod.getValue() )
-        || "Azure Active Directory - Integrated".equals( azureSqlDBJdbcAuthMethod.getValue() ) ) );
-    userNameBox.setDisabled( azureSqlDBJdbcAuthMethod != null && "Azure Active Directory - Integrated".equals( azureSqlDBJdbcAuthMethod.getValue() ) );
+    if ( azureSqlDBJdbcAuthMethod != null ) {
+      passwordBox.setDisabled(
+          "Azure Active Directory - Universal With MFA".equals( azureSqlDBJdbcAuthMethod.getValue() )
+        || "Azure Active Directory - Integrated".equals( azureSqlDBJdbcAuthMethod.getValue() ) );
+      userNameBox.setDisabled( "Azure Active Directory - Integrated".equals( azureSqlDBJdbcAuthMethod.getValue() ) );
+    }
```

Comment:
When `azureSqlDBJdbcAuthMethod == null`, shouldn't `passwordBox.setDisabled` and `userNameBox.setDisabled` still be being changed?
This way, they are not. Previously, they kind of were, despite the NPE.

Reply:
Hello @dcleao, thanks for your review. This solution is similar to logic inside of handleUseSecurityCheckbox method.

Reply:
Hello folks!
My comment is just a call to attention. If, for example, `passwordBox` and `userNameBox` are not visible when `azureSqlDBJdbcAuthMethod` is `null`, then the new code should be equivalent for all purposes. Otherwise, it would not.